### PR TITLE
don't merge yet, front end clean up and back end custom fields bug fix

### PR DESCRIPTION
**Pull request by jfrazier312**

### CODE REVIEW

File: app/controllers/attachments_controller.rb [hunk 1]
```diff
@@ -9,18 +11,31 @@ def new
 	end
 
 	def create
-		@attachment = Attachment.create( attachment_params )
-
-		if @attachment.save!
+		begin
+			@attachment = Attachment.new(attachment_params)
+			@attachment.save!
 			flash[:success] = "File uploaded!"
```
Comment (jfrazier312) [inline app/controllers/attachments_controller.rb]: @boyangles @duckgirl07  This is not rescuing the validation exception for some reason when you upload a non-PDF File.

@duckgirl07 When I click upload file with no file selected, it adds the docs/original/missing.png, which causes an error if you click on it

File: app/models/custom_field.rb [hunk 2]
```diff
@@ -32,7 +32,7 @@ class CustomField < ApplicationRecord
   after_update :clear_associated_field_entries, if: :field_type_changed?
   after_create {
     create_items_for_custom_fields(self.id)
-    create_stocks_for_custom_fields(self.id)
+    create_stocks_for_custom_fields(self.id) if self.is_stock
   }
```
Comment (jfrazier312) [inline app/models/custom_field.rb]: @boyangles This is the custom fields fix. poke around and make sure you think this works

File: app/views/requests/_editrequestitems.html.erb [hunk 3]
```diff
@@ -13,7 +13,7 @@
       <% item = Item.find(req_item.item_id) %>
       <tr>
         <td><%= link_to item.unique_name, item_path(item.id) %></td>
-        <td class="text-right">
+        <td class="text-center">
           <a href="#" data-xeditable="true"
```
Comment (jfrazier312) [inline app/views/requests/_editrequestitems.html.erb]: @duckgirl07 I changed these all to the center, I thought it looked better for when I added the serial tags column. If you don't like it or I'm missing some areas feel free to change them back

File: app/views/request_item_comments/_request_item_comment.html.erb [hunk 4]
```diff
@@ -1,12 +1,11 @@
-<div class="container">
-  <div class="row">
+<div class="container word-wrap break-word">
+  <div class="row word-wrap break-word">
     <div class="col-sm-2">
```
Comment (jfrazier312) [inline app/views/request_item_comments/_request_item_comment.html.erb]: @duckgirl07 this only breaks the text if it has spaces... if you do a long word with no breaks then it overflows?

File: app/controllers/attachments_controller.rb (hunk 1 above)
Comment (boyangles) [inline app/controllers/attachments_controller.rb]: Workaround using pre-validations. It seems like Paperclip is bypassing standard error workflow.

**Comment by duckgirl07** [comment]
i believe i've fixed the javascript issues popping up with quantity-change/requesting -- please teset!

**Pull request merged**